pkg/devices/api.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package devices
+
+import (
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+type apiDevice struct {
+	device        *pluginapi.Device
+	infoProviders []types.DeviceInfoProvider
+}
+
+// NewAPIDevice returns an interface of type APIDevice
+func NewAPIDevice(id string, infoProviders []types.DeviceInfoProvider) types.APIDevice {
+	return &apiDevice{
+		device: &pluginapi.Device{
+			ID:     id,
+			Health: pluginapi.Healthy,
+		},
+		infoProviders: infoProviders,
+	}
+}
+
+func (ad *apiDevice) GetDeviceSpecs() []*pluginapi.DeviceSpec {
+	dSpecs := make([]*pluginapi.DeviceSpec, 0)
+	for _, infoProvider := range ad.infoProviders {
+		dSpecs = append(dSpecs, infoProvider.GetDeviceSpecs()...)
+	}
+
+	return dSpecs
+}
+
+func (ad *apiDevice) GetEnvVal() string {
+	// Currently Device Plugin does not support returning multiple Env Vars
+	// so we use the value provided by the first InfoProvider.
+	return ad.infoProviders[0].GetEnvVal()
+}
+
+func (ad *apiDevice) GetMounts() []*pluginapi.Mount {
+	mnt := make([]*pluginapi.Mount, 0)
+	for _, infoProvider := range ad.infoProviders {
+		mnt = append(mnt, infoProvider.GetMounts()...)
```

Comment:
Do we need to check if `GetMounts` returns nil?

Reply:
For what? `append` works correctly with nil slice.